src/platform-aws.c

```diff
@@ -202,6 +202,10 @@ ncclResult_t platform_init(void)
 	 * To prevent NCCL based applications from getting aborted when using
 	 * fork(), the plugin explicitly enables FI_EFA_FORK_SAFE environment
 	 * variable, even in legacy environments where the overhead is high.
+	 *
+	 * The Neuron team has asked us to skip trying to set this
```

Comment:
Is the plan to set it in the command-line rather than letting the code do it? If that's the case, then we shouldn't be executing setenv at all thus avoiding the race.

Reply:
Their claim is that the XLA proxy configuration (which uses its own multi-threaded process) doesn't call fork, so this isn't a problem.  I mostly disagree with this assessment.  Nvidia NCCL tends not to have a bunch of threads at that point in the init, and this code has been there forever with Nvidia, so I think we should leave it for now.

Another option, which sucks, would be to link against rdma-core and only set this if on a kernel that isn't new enough to not need fork stuff.

Reply:
Have they tried testing their workload with this patch (and without setting the environment variable during launch) that they do not fail? It would be easy to check if their assessment is right.

Reply:
Talked internally and it will be handled by apps themselves.